pkg/scc/update.go

```diff
@@ -19,54 +19,184 @@ limitations under the License.
+	Config_submariner_operator_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-operator-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-operator
+  namespace: submariner-operator
+`
+	Config_submariner_routeagent_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-routeagent-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-routeagent
+  namespace: submariner-operator
+`
+	Config_submariner_globalnet_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-globalnet-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-globalnet
+  namespace: submariner-operator
+`
+	Config_submariner_networkplugin_syncer_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-networkplugin-syncer-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-networkplugin-syncer
+  namespace: submariner-operator
+`
+
+	Config_submariner_diagnose_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-diagnose-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-diagnose
+  namespace: submariner-operator
+`
+	Config_submariner_lighthouse_agent_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-lighthouse-agent-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-lighthouse-agent
+  namespace: submariner-operator
+`
+
+	Config_submariner_lighthouse_coredns_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-lighthouse-coredns-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-lighthouse-coredns
+  namespace: submariner-operator
+`
+)
+
+var (
+	privilegedClusterRoleDefined = false
+
+	nameToYaml = map[string]string{
```

Comment:
Once the yamls are added to the **submariner-operator** we can have the caller pass in the yaml name instead of using a hardcoded map.

Reply:
Hmm, not sure that I understood your comment,
can you explain how the appropriate CRB YAML file should be retrieved given the component name without using something like map ?

Reply:
You mentioned in the PR you defined the "YAML files just to get your feedback and to simplify testing, it should move to submariner-operator repo as the rest of the embedded YAML files.". Once we do that then the caller can just pass in the desired yaml name directly, eg `Config_submariner_gateway_privilged_scc_clusterrolebinding_yaml`, instead of maintaining a hardcoded mapping to get it indirectly.